core/src/main/java/jenkins/util/JenkinsJVM.java

```diff
@@ -0,0 +1,73 @@
+package jenkins.util;
+
+import hudson.WebAppMain;
+import javax.servlet.ServletContextEvent;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * A utility class to identify if the current JVM is the one that is running {@link Jenkins}
```

Comment:
In Jenkins 4.0 it may be that the "master" is not the one running `Jenkins` so I am being clear that this is the one that actually has a `Jenkins` singleton instance